src/test/rpc/ValidatorRPC_test.cpp

```diff
@@ -0,0 +1,334 @@
+
+        // Configure environment with single validator site and publisher key
+        Env env{
+            *this,
+            envconfig([&](std::unique_ptr<Config> cfg) {
+                cfg->section(SECTION_VALIDATOR_LIST_SITES).append(siteURI);
+                cfg->section(SECTION_VALIDATOR_LIST_KEYS)
+                    .append(strHex(publisherPublic));
+                return cfg;
+            }),
+        };
+
+        //----------------------------------------------------------------------
+        // Publisher list manifest has not been loaded yet
+        {
+            auto const jrr = env.rpc("server_info")[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::info][jss::validator_list_expires] == "unknown");
+        }
+        {
+            auto const jrr = env.rpc("validator_lists")[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::validation_quorum].asUInt() ==
+                std::numeric_limits<std::uint32_t>::max());
+            BEAST_EXPECT(jrr[jss::validator_keys].size() == 0);
+            BEAST_EXPECT(jrr[jss::validator_list_expires] == "unknown");
+
+            if (BEAST_EXPECT(jrr[jss::publisher_lists].size() == 1))
+            {
+                auto jp = jrr[jss::publisher_lists][0u];
+                BEAST_EXPECT(jp[jss::available] == false);
+                BEAST_EXPECT(jp[jss::list].size() == 0);
+                BEAST_EXPECT(jp[jss::seq].asUInt() == 0);
+                BEAST_EXPECT(
+                    jp[jss::pubkey_publisher] == strHex(publisherPublic));
+                BEAST_EXPECT(
+                    jp[jss::expiration] == to_string(TimeKeeper::time_point{}));
+            }
+        }
+        {
+            auto const jrr = env.rpc("validator_sites")[jss::result];
+            if (BEAST_EXPECT(jrr[jss::validator_sites].size() == 1))
+            {
+                auto js = jrr[jss::validator_sites][0u];
+                BEAST_EXPECT(js[jss::refresh_interval_min].asUInt() == 5);
+                BEAST_EXPECT(js[jss::uri] == siteURI);
+                BEAST_EXPECT(!js.isMember(jss::last_refresh_time));
+                BEAST_EXPECT(!js.isMember(jss::last_refresh_status));
+            }
+        }
+
+        //----------------------------------------------------------------------
+        // Load the list from the site and check that the published keys are
+        // reported
+
+        auto const now = std::chrono::steady_clock::now();
+        auto& ioService = env.app().getIOService();
+        NetClock::time_point const expiration = env.timeKeeper().now() + 3600s;
+        // Setup the trusted publisher server to publish the list of validator
+        // keys
+        TrustedPublisherServer server(
+            ep,
+            ioService,
+            publisherSigningKeys,
+            manifest,
+            1,
+            expiration,
+            1,
+            keys);
+        env.app().validatorSites().start();
```

Comment:
I'm getting the same test failures as Travis:
```
#32 failed: ValidatorRPC_test.cpp(269)
#33 failed: ValidatorRPC_test.cpp(273)
#34 failed: ValidatorRPC_test.cpp(275)
#35 failed: ValidatorRPC_test.cpp(277)
#37 failed: ValidatorRPC_test.cpp(290)
#38 failed: ValidatorRPC_test.cpp(291)
#39 failed: ValidatorRPC_test.cpp(301)
#41 failed: ValidatorRPC_test.cpp(304)
#45 failed: ValidatorRPC_test.cpp(314)
#46 failed: ValidatorRPC_test.cpp(317)
```
The site isn't being fetched here because `ValidatorSite::start` was already called when `env` was constructed.
The only workaround I've thought of is to use a new `Env` for this set of tests (and construct it after the site is up).